unison-syntax/test/Unison/Test/Doc.hs

```diff
@@ -147,6 +150,8 @@ t s expected =
                   crash "actual != expected"
       )
     $ P.runParser (DP.doc (Name.toText . HQ'.toName . snd <$> typeOrTerm) (P.manyTill P.anySingle) P.eof) "test case" s
+  where
+    simpleScope = fromMaybe "default" $ headMay $ words $ filter isAlphaNum s
```

Comment:
I guess the issue is that some of the test names are not easy (and maybe not possible?) to provide at the command-line, due to weird stuff like newlines. And are there maybe tests with no name at all?

But could you add some comments to the code about what this line is trying to accomplish and why?